apisix/plugins/authz-keycloak.lua

```diff
@@ -17,15 +17,18 @@
 local core      = require("apisix.core")
 local http      = require "resty.http"
 local sub_str   = string.sub
-local url       = require "net.url"
-local tostring  = tostring
+local type      = type
 local ngx       = ngx
+local cjson     = require("cjson")
+local cjson_s   = require("cjson.safe")
```

Comment:
It is better to use the public methods provided here.[https://github.com/apache/apisix/blob/master/apisix/core/json.lua ](https://github.com/apache/apisix/blob/master/apisix/core/json.lua)

Reply:
Yep, using that one now. We're incurring a slight performance penalty by now using `cjson.safe.decode` through `apisix.core.json.decode` to decode values from the shared memory which are known to have been encoded correctly. But not sure how much of a hit that actually is. Just this line here: https://github.com/jenskeiner/apisix/blob/26277f1ea98366ac852ee6673667e1be1320001c/apisix/plugins/authz-keycloak.lua#L190